flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobClient.java

```diff
@@ -130,7 +130,12 @@ static void downloadFromBlobServer(
             throws IOException {
 
         final byte[] buf = new byte[BUFFER_SIZE];
-        LOG.info("Downloading {}/{} from {}", jobId, blobKey, serverAddress);
+        LOG.info(
+                "Downloading {}/{} from {} (retry {})",
```

Comment:
I think the `retry` is not needed here. If a retry happens, the retry attempt is already logged (in the code below).
Besides that, `numFetchRetries` is not the current retry attempt but is the max retry count.